Redfish Exporter v2.x

jovial opened the pull request.

technowhizz committed:
Restore old UUID of the redfish dashboard

The redfish dashboard was updated to use a new UUID, however this caused
errors in the Grafana logs and Grafana didn't update the dashboard.
Reverting this to the old UUID fixes the issue.

m-bull committed:
Configuration for the StackHPC fork of Redfish Exporter

jovial committed:
Update redfish dashboard

- Compatability with Dell servers
- Added health round up panels
- Add variables for different groups e.g overcloud vs compute

jovial committed:
Increase query lookback delta

Prometheus will regard metrics collected over this perioid as stale as
such they won't show up in grafana.

jovial commented on etc/kayobe/kolla/globals.yml:
```diff
@@ -26,7 +26,7 @@ kolla_image_tags:
 
 opensearch_heap_size: 8g
-prometheus_cmdline_extras: "--storage.tsdb.retention.time=30d"
+prometheus_cmdline_extras: "--storage.tsdb.retention.time=30d --query.lookback-delta=15m"
```
We could something fancier like twice the longest scrape interval.

jovial committed:
Add raw tags

m-bull committed:
Re-tweak Redfish dashboard for Lenovo

jovial committed:
Make query lookback delta setting smarter

jovial committed:
Actually set lookback delta

JohnGarbutt commented on etc/kayobe/stackhpc-monitoring.yml:
```diff
@@ -1,5 +1,9 @@
+###############################################################################
+# Prometheus server configuration
+
+stackhpc_prometheus_query_lookback_delta: "{{ [redfish_exporter_scrape_interval, stackhpc_os_capacity_scrape_interval, 300] | max }}s"
```
Do you need to add a few seconds here to avoid some of the edge effects you saw in testing?

JohnGarbutt replied:
I am also wonder if we need to change the openstack exporter threshold to be included in here as well. For larger clouds, its probably a bit too aggressive by default.

jovial replied:
I've added 30 seconds to give some leeway. Have also added an openstack exporter scrape interval variable and added it the max calculation.

JohnGarbutt commented on etc/kayobe/kolla/globals.yml:
```diff
@@ -26,7 +26,7 @@ kolla_image_tags:
 
 opensearch_heap_size: 8g
-prometheus_cmdline_extras: "--storage.tsdb.retention.time=30d"
+prometheus_cmdline_extras: "--storage.tsdb.retention.time=30d --query.lookback-delta={{ stackhpc_prometheus_query_lookback_delta }}"
```
FWIW, I am +1 tweaking this, given we regularly increase the interval for various exporters.

jovial committed:
Adds openstack exporter scrape interval

jovial committed:
Add some raw tags

jovial committed:
Slightly less offensive variant on raw tags

jovial committed:
Refresh release note

jovial committed:
Update dashboard

Alex-Welsh commented on releasenotes/notes/bumps-redfish-exporter-to-v2-11032fb9dde36283.yaml:
```diff
@@ -0,0 +1,19 @@
+---
+features:
+  - Upgrades the redfish exporter container image to the v2.x series.
```
Is this change missing a new image tag?

Alex-Welsh replied:
Nevermind, I've just realised this is targeting another feature branch

Alex-Welsh commented on releasenotes/notes/bumps-redfish-exporter-to-v2-11032fb9dde36283.yaml:
```diff
@@ -0,0 +1,19 @@
+    ``stackhpc_prometheus_openstack_exporter_interval``
+    configuration variables.
+fixes:
+  - |
+    Sets the prometheus server side option ``query.lookback-delta`` to
+    the largest scrape interval so that metrics are not from exporters
+    with large scrape intervals are not marked stale before the next scrape.
```
```suggestion
  - |
    Sets the prometheus server side option ``query.lookback-delta`` to
    the largest scrape interval so that metrics from exporters with large
    scrape intervals are not marked stale before the next scrape.
```

Alex-Welsh commented on etc/kayobe/stackhpc-monitoring.yml:
```diff
@@ -45,16 +54,25 @@ stackhpc_os_capacity_openstack_cacert: ""
 stackhpc_enable_redfish_exporter: false
 
+# How often to scrape the BMCs in seconds.
+stackhpc_redfish_exporter_scrape_interval: "{{ [8 * groups['redfish_exporter_targets'] | length, 300] | max }}"
```
Where does this get applied?

MoteHue replied:
Fixed to apply this now

MoteHue commented on etc/kayobe/kolla/config/prometheus/prometheus.yml.d/60-redfish.yml:
```diff
@@ -7,7 +7,7 @@ scrape_configs:
     metrics_path: /redfish
     scrape_timeout: 300s
-    scrape_interval: {{ [8 * groups['redfish_exporter_targets'] | length, 300] | max }}s
+    scrape_interval: {{ redfish_exporter_scrape_interval }}s
```
```suggestion
    scrape_interval: {{ stackhpc_redfish_exporter_scrape_interval }}s
```

MoteHue committed:
Apply suggestions from code review

Co-authored-by: Alex-Welsh <112560678+Alex-Welsh@users.noreply.github.com>

MoteHue committed:
Bump to redfish exporter v2.1.1-stackhpc

MoteHue committed:
Temporarily remove redfish log collection

This isn't working yet (takes far too long to scrape),let's
split it out into a separate patch.

Also removed the query lookback delta config, as this was
working around the much longer scrape times.

MoteHue commented:
We've decided to push the log collection out for now, to be proposed in a new PR later. I'll leave the conversation threads open here so they can be seen in that work.

MoteHue committed:
Merge remote-tracking branch 'origin/stackhpc/2024.1' into bugfix/edfish-exporter-2.0/dell

dougszumski reviewed (approved):
Let's merge and follow up with anything else

MoteHue commented:
> We've decided to push the log collection out for now, to be proposed in a new PR later. I'll leave the conversation threads open here so they can be seen in that work.

Need to resolve the convos to merge :(
